website/docs/r/vpc_security_group_egress_rule.html.markdown

```diff
@@ -26,12 +26,15 @@ resource "aws_vpc_security_group_egress_rule" "example" {
   cidr_ipv4   = "10.0.0.0/8"
   from_port   = 80
   ip_protocol = "tcp"
-  to_port     = 8080
+  to_port     = 80
 }
 ```
 
 ## Argument Reference
 
+~> **NOTE on optional/required attributes:** Although `cidr_ipv4`, `cidr_ipv6`, `prefix_list_id`, and `referenced_security_group_id` are all marked as optional, you *must* provide one of them in order to configure the destination of the traffic.
+`from_port` and `to_port` are required, unless `ip_protocol` is set to `-1` or `icmpv6`.
```

Comment:
- Restructured this a bit to concatenate the paragraphs (I'm not _certain_ that breaking to a second paragraph would do funny things with the `~>`, but I think it's worth ensuring it). 
- Added a couple of words to make the last sentence feel more complete. This might be my own preferences more than anything, but I tend to avoid starting the sentence with preformatted text 🤷 .
- Removed the bit after the `Note` (which I formatted according to the [callouts docs](https://developer.hashicorp.com/terraform/registry/providers/docs#callouts); I don't think we typically add any additional text around `Note`, so keeping with precedence. I'm not _super_ adamant about this, so I'm open to input if you feel like it's worth keeping.

```suggestion
~> **Note** Although `cidr_ipv4`, `cidr_ipv6`, `prefix_list_id`, and `referenced_security_group_id` are all marked as optional, you *must* provide one of them in order to configure the destination of the traffic. The `from_port` and `to_port` arguments are required unless `ip_protocol` is set to `-1` or `icmpv6`.
```